rune/libenclave/attestation/sgx/ias/ias.go

```diff
@@ -431,25 +441,37 @@ func checkVerificationReport(resp *http.Response, quote []byte, nonce string) (*
 
 	var x509Certs []*x509.Certificate
 	if x509Certs, err = x509.ParseCertificates(derCerts); err != nil {
-		return status, fmt.Errorf("Failed to parse certificates: %s", err)
+		return status, nil, fmt.Errorf("Failed to parse certificates: %s", err)
 	}
 
 	cert := x509Certs[0]
 	if err = cert.CheckSignature(x509.SHA256WithRSA, rawReport, sig); err != nil {
-		return status, fmt.Errorf("Failed to verify the attestation verification report: %s",
+		return status, nil, fmt.Errorf("Failed to verify the attestation verification report: %s",
 			err)
 	}
 
 	for _, parentCert := range x509Certs[1:] {
 		if err = cert.CheckSignatureFrom(parentCert); err != nil {
-			return status, fmt.Errorf("Failed to verify the certificate (%s) with parent certificate (%s): %s",
+			return status, nil, fmt.Errorf("Failed to verify the certificate (%s) with parent certificate (%s): %s",
 				cert.Subject.String(), parentCert.Subject.String(), err)
 		}
 
 		cert = parentCert
 	}
 
-	return status, nil
+	iasReport := make(map[string]string)
+
+	iasReport["StatusCode"] = strconv.FormatUint(uint64(resp.StatusCode), 10)
+	iasReport["Request-ID"] = resp.Header.Get("Request-ID")
+	iasReport["X-Iasreport-Signature"] = resp.Header.Get("X-Iasreport-Signature")
+	iasReport["X-Iasreport-Signing-Certificate"] = resp.Header.Get("X-Iasreport-Signing-Certificate")
+	iasReport["ContentLength"] = strconv.FormatUint(uint64(resp.ContentLength), 10)
+	iasReport["Content-Type"] = resp.Header.Get("Content-Type")
+
+	respBody := string(rawReport)
+	iasReport["Body"] = respBody
```

Comment:
This function should not do this. iasReport can be created as long as resp is available right? So you should write a new function called `formatIasReport(resp) map[string]` and caller should invoke this function after calling checkVerificationReport().